akka-http/src/main/scala/akka/http/javadsl/server/directives/RouteDirectives.scala

```diff
@@ -239,47 +239,56 @@ abstract class RouteDirectives extends RespondWithDirectives {
    */
   @CorrespondsTo("complete")
   def completeWithFuture(value: CompletionStage[HttpResponse]) = RouteAdapter {
-    D.complete(value.asScala.fast.map(_.asScala))
+    D.complete(value.asScala.fast.map(_.asScala).recover(unwrapCompletionException))
   }
 
   /**
    * Completes the request by marshalling the given future value into an http response.
    */
   @CorrespondsTo("complete")
   def completeOKWithFuture(value: CompletionStage[RequestEntity]) = RouteAdapter {
-    D.complete(value.asScala.fast.map(_.asScala))
+    D.complete(value.asScala.fast.map(_.asScala).recover(unwrapCompletionException))
   }
 
   /**
    * Completes the request by marshalling the given future value into an http response.
    */
   @CorrespondsTo("complete")
   def completeOKWithFutureString(value: CompletionStage[String]) = RouteAdapter {
-    D.complete(value.asScala)
+    D.complete(value.asScala.recover(unwrapCompletionException))
   }
 
   /**
    * Completes the request using the given future status code.
    */
   @CorrespondsTo("complete")
   def completeWithFutureStatus(status: CompletionStage[StatusCode]): Route = RouteAdapter {
-    D.complete(status.asScala.fast.map(_.asScala))
+    D.complete(status.asScala.fast.map(_.asScala).recover(unwrapCompletionException))
   }
 
   /**
    * Completes the request with an `OK` status code by marshalling the given value into an http response.
    */
   @CorrespondsTo("complete")
   def completeOKWithFuture[T](value: CompletionStage[T], marshaller: Marshaller[T, RequestEntity]) = RouteAdapter {
-    D.complete(value.asScala.fast.map(v ⇒ ToResponseMarshallable(v)(fromToEntityMarshaller()(marshaller))))
+    D.complete(value.asScala.fast.map(v ⇒ ToResponseMarshallable(v)(fromToEntityMarshaller()(marshaller))).recover(unwrapCompletionException))
   }
 
   /**
    * Completes the request by marshalling the given value into an http response.
    */
   @CorrespondsTo("complete")
   def completeWithFuture[T](value: CompletionStage[T], marshaller: Marshaller[T, HttpResponse]) = RouteAdapter {
-    D.complete(value.asScala.fast.map(v ⇒ ToResponseMarshallable(v)(marshaller)))
+    D.complete(value.asScala.fast.map(v ⇒ ToResponseMarshallable(v)(marshaller)).recover(unwrapCompletionException))
+  }
+
+  // TODO: This might need to be raised as an issue to scala-java8-compat instead.
```

Comment:
:+1: Can you file an issue there as well?

Reply:
https://github.com/scala/scala-java8-compat/issues/120